File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/menu/handlers/ViewFilesClickHandler.java
package com.lasthopesoftware.bluewater.client.library.items.menu.handlers;

import android.view.View;
import com.lasthopesoftware.bluewater.client.library.items.IItem;
import com.lasthopesoftware.bluewater.client.library.items.menu.NotifyOnFlipViewAnimator;

import static com.lasthopesoftware.bluewater.client.library.items.media.files.list.FileListActivity.startFileListActivity;

/**
 * Created by david on 4/3/15.
 */
public final class ViewFilesClickHandler extends AbstractMenuClickHandler {
    private final IItem item;

    public ViewFilesClickHandler(NotifyOnFlipViewAnimator menuContainer, IItem item) {
        super(menuContainer);
        this.item = item;
    }

    @Override
    public void onClick(View v) {
        startFileListActivity(v.getContext(), item);
        super.onClick(v);
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/items/menu/handlers/ViewFilesClickHandler.kt
package com.lasthopesoftware.bluewater.client.browsing.items.menu.handlers

import android.view.View
import com.lasthopesoftware.bluewater.client.browsing.items.IItem
import com.lasthopesoftware.bluewater.client.browsing.items.menu.NotifyOnFlipViewAnimator
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.browsing.startItemBrowserActivity

class ViewFilesClickHandler(menuContainer: NotifyOnFlipViewAnimator, private val libraryId: LibraryId, private val item: IItem) : AbstractMenuClickHandler(menuContainer) {
    override fun onClick(v: View) {
        v.context.startItemBrowserActivity(libraryId, item)
        super.onClick(v)
    }
}
